tests/components/axis/test_config_flow.py

```diff
@@ -77,15 +74,44 @@ def mock_constructor(loop, host, username, password, port, web_proto):
     with patch(
         "homeassistant.components.axis.config_flow.get_device",
         return_value=mock_coro(mock_device),
-    ), pytest.raises(homeassistant.data_entry_flow.AbortFlow):
-        await flow.async_step_user(
+    ):
+        result = await flow.async_step_user(
             user_input={
                 config_flow.CONF_HOST: "1.2.3.4",
                 config_flow.CONF_USERNAME: "user",
                 config_flow.CONF_PASSWORD: "pass",
                 config_flow.CONF_PORT: 80,
             }
         )
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+
+async def test_flow_manual_input_updates_existing_configuration(hass):
+    """Test that config flow fails on already configured device."""
+    await setup_axis_integration(hass)
+
+    flow = config_flow.AxisFlowHandler()
+    flow.hass = hass
+    flow.context = {}
+
+    mock_device = Mock()
+    mock_device.vapix.params.system_serialnumber = MAC
+
+    with patch(
+        "homeassistant.components.axis.config_flow.get_device",
+        return_value=mock_coro(mock_device),
+    ):
+        result = await flow.async_step_user(
```

Comment:
It's better to use the core interfaces in the test. It will make tests more robust.

```py
result = await hass.config_entries.flow.async_init(
    DOMAIN, context={"source": "user"}, data={...}
)
```

https://github.com/home-assistant/home-assistant/blob/63347ebeb5e84b5d1fbeeb25c5017831a2ff40bc/homeassistant/data_entry_flow.py#L94-L96

Reply:
Done